diff --git a/Source/Teensy/MinimalBoot/Common/NativeNES/vendor/chips/m6502.h b/Source/Teensy/MinimalBoot/Common/NativeNES/vendor/chips/m6502.h
--- a/Source/Teensy/MinimalBoot/Common/NativeNES/vendor/chips/m6502.h
+++ b/Source/Teensy/MinimalBoot/Common/NativeNES/vendor/chips/m6502.h
@@ -203,6 +203,9 @@
 #include <stdint.h>
 #include <stdbool.h>

+#ifndef M6502_CODE
+#define M6502_CODE
+#endif
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -349,8 +353,8 @@ typedef struct {
 } m6502_t;
 /* initialize a new m6502 instance and return initial pin mask */
-uint64_t m6502_init(m6502_t* cpu, const m6502_desc_t* desc);
+M6502_CODE uint64_t m6502_init(m6502_t* cpu, const m6502_desc_t* desc);
 /* execute one tick */
-uint64_t m6502_tick(m6502_t* cpu, uint64_t pins);
+M6502_CODE uint64_t m6502_tick(m6502_t* cpu, uint64_t pins);
 /* perform m6510 port IO (only call this if M6510_CHECK_IO(pins) is true) */
 uint64_t m6510_iorq(m6502_t* cpu, uint64_t pins);
 // prepare m6502_t snapshot for saving
@@ -621,6 +625,6 @@ static inline void _m6502_sbx(m6502_t* cpu, uint8_t v) {
 }
 #undef _M6502_NZ
-uint64_t m6502_init(m6502_t* c, const m6502_desc_t* desc) {
+M6502_CODE uint64_t m6502_init(m6502_t* c, const m6502_desc_t* desc) {
     CHIPS_ASSERT(c && desc);
     memset(c, 0, sizeof(*c));
     c->P = M6502_ZF;
@@ -713,5 +717,5 @@ void m6502_set_p(m6502_t* c, uint8_t v) { CHIPS_ASSERT(c); c->P = v; }
 #pragma warning(disable:4244)   /* conversion from 'uint16_t' to 'uint8_t', possible loss of data */
 #endif
-uint64_t m6502_tick(m6502_t* c, uint64_t pins) {
+M6502_CODE uint64_t m6502_tick(m6502_t* c, uint64_t pins) {
     if (pins & (M6502_SYNC|M6502_IRQ|M6502_NMI|M6502_RDY|M6502_RES)) {
         // interrupt detection also works in RDY phases, but only NMI is "sticky"
